### PR TITLE
rename kubeapps-dashboard svc to sound more internal

### LINKED ISSUE #602: Rename kubeapps-dashboard Service to avoid confusion over what Service to use
A lot of users (#382, https://github.com/kubeapps/kubeapps/issues/577#issuecomment-419392268) mistake the `kubeapps-dashboard` Service as the one to set as a LoadBalancer or proxy to using an Ingress resource, when in fact the `kubeapps` Service should be used. It doesn't help that the default Kubernetes Dashboard manifest calls their Service `kubernetes-dashboard`.

We should rename this Service to something that sounds like less of an entrypoint, for example:

- kubeapps-react
- kubeapps-static
- kubeapps-internal

This should help avoid users mistaking the correct Service in the future.

### PR DESCRIPTION
**Pull request by prydonius**
In order to avoid confusion around the `kubeapps-dashboard` and
`kubeapps` Service, we should rename the `kubeapps-dashboard` Service
to sound like more of an internal component rather than a possible
entrypoint to the dashboard.

fixes #602

### CODE REVIEW

File: chart/kubeapps/templates/_helpers.tpl [hunk 1]
```diff
@@ -79,14 +79,14 @@ Create name for the chartsvc based on the fullname
 Create name for the dashboard based on the fullname
 */}}
 {{- define "kubeapps.dashboard.fullname" -}}
```
Comment (prydonius) [inline chart/kubeapps/templates/_helpers.tpl]: the fact that the values and the templates are still called dashboard might be confusing, so we might want to change these too? What do you think?

File: chart/kubeapps/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: kubeapps
-version: 0.4.0
+version: 0.4.1
```
Comment (arapulido) [inline chart/kubeapps/Chart.yaml]: You will need to rebase before merging, as there is another pending update with that version number.

File: chart/kubeapps/templates/_helpers.tpl [hunk 3]
```diff
@@ -79,14 +79,14 @@ Create name for the chartsvc based on the fullname
...
-{{ template "kubeapps.fullname" . }}-dashboard
+{{ template "kubeapps.fullname" . }}-react
 {{- end -}}
 
 {{/*
 Create name for the dashboard config based on the fullname
 */}}
 {{- define "kubeapps.dashboard-config.fullname" -}}
-{{ template "kubeapps.fullname" . }}-dashboard-config
+{{ template "kubeapps.fullname" . }}-react-config
```
Comment (arapulido) [inline chart/kubeapps/templates/_helpers.tpl]: mmm, not a fan of calling it with just the technology. maybe `reactapp` ?

File: chart/kubeapps/templates/_helpers.tpl (hunk 1 above)
Comment (andresmgot) [reply chart/kubeapps/templates/_helpers.tpl]: If we rename it to `kubeapps-dashboard-internal` the templates would still make sense. I wouldn't call it `react` to not make it depend on the underlying technology.

File: chart/kubeapps/templates/_helpers.tpl (hunk 3 above)
Comment (migmartri) [reply chart/kubeapps/templates/_helpers.tpl]: Yeah, me neither, naming is hard though because I can not think in a good name for this.

**Comment by prydonius** [comment]
Discussed with @migmartri and he suggested to prefix all the names with `internal` except the main proxy Service, which should hopefully give a better indication of which service to use.

I will also modify the values.yaml file to put more emphasis on some of the options (e.g. frontend) over the others.

**Comment by prydonius** [comment]
@migmartri @andresmgot  I've made the described changes, PTAL!

screen shot 2018-09-11 at 14 04 42

screen shot 2018-09-11 at 14 05 43

**Comment by prydonius** [comment]
I tested it by upgrading my existing installation, so upgrades seem to work.

**Comment by migmartri** [comment]
> I would still write a short document or a section in the getting-started or dashboard documents to explain how to expose Kubeapps.

We already have how to port forward both in the readme and getting started, or you mean how to expose it via loadbalancer, ingress and so on?

**Comment by prydonius** [comment]
I think for now, the configuration options should just be in the chart readme. Once we have a docs site, we can rethink this and have more pages for configuration like this.

**Pull request merged**